### PR TITLE
Add GitPod as an online tool

### PR DESCRIPTION
**Pull request by LinuxSuRen**

### CODE REVIEW

File: .github/workflows/Dockerfile-ks-jenkins-cache [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM maven:3.8.1-jdk-11-openj9 as cache
+
+# Cache the maven local repository
+RUN curl -L https://github.com/LinuxSuRen/http-downloader/releases/download/v0.0.29/hd-linux-amd64.tar.gz | tar xzv && \
+    mv hd /usr/local/bin && \
+    hd install jenkins-zh/jenkins-cli/jcli && \
+    apt update -y && \
+    apt install git make -y && \
+    git clone https://github.com/kubesphere/ks-jenkins && \
+    cd ks-jenkins && \
```
Comment (JohnNiang) [inline .github/workflows/Dockerfile-ks-jenkins-cache:10]: Here could be simplified by:

```shell
jcli cwp --install-artifacts --config-path ks-jenkins/formula.yaml || true && \
rm -rf ks-jenkins
```

File: .gitpod.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    jcli cwp --install-artifacts --config-path formula.yaml || true && \
+    cd .. && rm -rf ks-jenkins
+
+FROM gitpod/workspace-full
+
+USER root
+COPY --from=cache /usr/local/bin/hd /usr/local/bin/hd
+COPY --from=cache /root/.m2/repository /home/gitpod/.m2/repository
+
+USER gitpod
```
Comment (JohnNiang) [inline .gitpod.Dockerfile:15]: This declaration may be redundant.

File: .github/workflows/Dockerfile-ks-jenkins-cache (hunk 1 above)
Comment (LinuxSuRen) [reply .github/workflows/Dockerfile-ks-jenkins-cache:10]: Yes

**Pull request merged**